src/dreyfus_index_manager.erl

```diff
@@ -113,25 +115,48 @@ handle_db_event(DbName, created, _St) ->
 handle_db_event(DbName, deleted, _St) ->
     gen_server:cast(?MODULE, {cleanup, DbName}),
     {ok, nil};
+handle_db_event(DbName, {ddoc_updated, DDocId}, _St) ->
```

Comment:
You are subscribing for `all_dbs` [here](https://github.com/cloudant-labs/dreyfus/blob/0739e2329877ea9222c8b00340c9d3b37b423220/src/dreyfus_index_manager.erl#L48). This means that you would get non-clustered databases as well. You should distinguish these two cases. Which can be done by matching on `<<"shards/", _/binary>> = DbName`.